spec/factories/miq_widget_set.rb

```diff
@@ -2,5 +2,21 @@
   factory :miq_widget_set do
     sequence(:name)         { |n| "widget_set_#{seq_padded_for_sorting(n)}" }
     sequence(:description)  { |n| "widget_set_#{seq_padded_for_sorting(n)}" }
+
+    transient do
+      widget_id             { nil }
+      last_group_db_updated { nil }
+    end
+
+    after(:build) do |widget_set, env|
+      unless widget_set.set_data
+        widget_set.valid?
+        widget_set.set_data[:col1] << (env.widget_id || FactoryBot.create(:miq_widget).id)
+      end
+
+      unless widget_set.owner_id
+        widget_set.owner = User.current_user || FactoryBot.create(:user)
```

Comment:
This seems like it is wrong.  Pretty sure this should be a `MiqGroup` object, and not a user, based on similar calls to it just below:

https://github.com/ManageIQ/manageiq/pull/20974/commits/f9f8fa4e87d571901bc367786f26471ef9637c47#diff-198f0b8c1a90286a1547ecd5a5d29fb7d26c125d2cb305b2a7d07f68c0edc55aL320

Further more, you also need to make sure you are defaulting `group` as well if you are going to go with this route.